app/models/check_availability_task.rb

```diff
@@ -18,4 +19,13 @@ def dispatch
     reload
     source.update!(:availability_message => "Message sent to RHC #{controller_message_id}") if status == 'ok'
   end
+
+  def timed_out?
+    time_interval = ClowderConfig.instance["CHECK_AVAILABILITY_TIMEOUT"] * 60 # in seconds
```

Comment:
@hsong-rh Can we make this a class variable in all the sub task classes and then use it from the timed_out? method in the base Task class.